chat_shell/chat_shell/tools/builtin/knowledge_base.py

```
@@ -143,6 +461,16 @@ async def _arun(
                     {"error": "No knowledge bases configured for this conversation."}
                 )
 
+            # Step 1: Check call limits BEFORE executing search
+            max_calls, exempt_calls = self._get_kb_limits()
```

Ruff (0.14.14)

Unpacked variable `exempt_calls` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)